### PR TITLE
feat(devtunnel): support geodistributed tunnels

### PR DESCRIPTION
**Pull request by coadler**

### CODE REVIEW

File: coderd/devtunnel/servers.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		Nodes: []TunnelNode{
+			{
+				ID:                1,
+				HostnameHTTPS:     "pit-1.try.coder.app",
+				HostnameWireguard: "pit-1.try.coder.app",
+				WireguardPort:     55551,
+			},
+		},
+	},
+}
```
Comment (deansheather) [inline coderd/devtunnel/servers.go:43]: It'd be nice if eventually this came from a server with a list of regions.

File: coderd/devtunnel/servers.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+var TunnelRegions = []TunnelRegion{
+	{
+		ID:           1,
+		LocationName: "US East Pittsburgh",
+		Nodes: []TunnelNode{
+			{
+				ID:                1,
+				HostnameHTTPS:     "pit-1.try.coder.app",
+				HostnameWireguard: "pit-1.try.coder.app",
+				WireguardPort:     55551,
```
Comment (deansheather) [inline coderd/devtunnel/servers.go:39]: These two fields could be returned by the server instead when it registers

**Review by deansheather — APPROVED** [review]

File: coderd/devtunnel/servers.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+			},
+		},
+	},
+}
+
+func PickTunnelNode() (TunnelNode, error) {
```
Comment (kylecarbs) [inline coderd/devtunnel/servers.go:45]: I might include the picking strategy when connecting to a node. `FindClosestNode` or something to that effect.

**Review by kylecarbs — APPROVED** [review]

File: coderd/devtunnel/tunnel.go [hunk 4]
```diff
@@ -227,6 +271,15 @@ func readOrGenerateConfig() (Config, error) {
...
 	}
 
+	if cfg.Version == 0 {
+		cfg.Tunnel = TunnelNode{
+			ID:                0,
+			HostnameHTTPS:     "wg-tunnel.coder.app",
+			HostnameWireguard: "wg-tunnel-udp.coder.app",
+			WireguardPort:     55555,
+		}
+	}
```
[the remark this answers is not in the document]
Comment (coadler) [reply coderd/devtunnel/tunnel.go:281]: Yeah, this is the old tunnel config. Eventually I'll force people to upgrade.

**Pull request merged**